modules/admin-ui/src/main/java/org/opencastproject/adminui/usersettings/UserSettingsService.java

```diff
@@ -78,6 +86,7 @@ public class UserSettingsService {
   @Activate
   public void activate(ComponentContext cc) {
     logger.info("Activating persistence manager for user settings");
+    db = dbSessionFactory.createSession(emf);
```

Comment:
Why initializing this in here? Couldn't we just turn this into an OSGi service which binds itself to the EntityManagerFactory?

Reply:
I thought about that, but I wasn't sure if every instance needs a different persistent unit, e.g. `AwsAssetDatabaseImpl`:

```java
@Reference(target = "(osgi.unit.name=org.opencastproject.assetmanager.aws.persistence)")
public void setEntityManagerFactory(EntityManagerFactory emf) {
  this.emf = emf;
}
```

Are `EntityManagerFactory` actually the same between modules? Wouldn't this require a `persistence.xml` in one module that contains every DTO?